File: Decorator/src/main/java/starbuzz_coffee/Chocolate.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package starbuzz_coffee;

/**
 *
 * @author crazy
 */
public class Chocolate extends Condimento {    
    public Chocolate(){
        costo = 20;
        nombre = "chocolate";
    }
    
    public Chocolate(Condimento wrappee){
        super(wrappee);
        costo = 20;
        nombre = "chocolate";
    }
    
    
}

File: DecoratorCpp/starbuzz/Chocolate.cpp
#include "Chocolate.h"

Chocolate::Chocolate() : Condimento() {
	costo = 40.f;
	nombre = "Chocolate";
}

Chocolate::Chocolate(Condimento* wrappee) : Condimento(wrappee) {
	costo = 40.f;
	nombre = "Chocolate";
}